DoxyFiles/man/man3/Elovich.3
.TH "Elovich" 3 "Segunda, 3 de Outubro de 2022" "Version 1.0.0" "Isotherm++" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Elovich \- Classe com as equacoes da isoterma de \fBElovich\fP  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Elovich\&.h>\fP
.PP
Subclasse de \fBTwoParameters\fP e \fBIsothermTemplate< Elovich >\fP\&.
.SS "Membros Públicos"

.in +1c
.ti -1c
.RI "\fBDefineIdentity\fP ('Elovich', ID::Elovich)"
.br
.RI "Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. "
.ti -1c
.RI "\fBElovich\fP ()=default"
.br
.RI "Construtora default\&. "
.ti -1c
.RI "\fBElovich\fP (const \fBElovich\fP &_orig)=default"
.br
.RI "Construtora de copia\&. "
.ti -1c
.RI "\fB~Elovich\fP () override=default"
.br
.RI "Destrutora\&. "
.ti -1c
.RI "\fBElovich\fP (const Real &_qmax, const Real &_k1)"
.br
.RI "Construtora com os parametros que definem a isoterma de \fBElovich\fP\&. "
.ti -1c
.RI "\fBElovich\fP & \fBoperator=\fP (const \fBElovich\fP &_orig)=default"
.br
.RI "Sobrecarga do operador =\&. "
.ti -1c
.RI "Real \fBQmax\fP () const"
.br
.RI "Funcao que informa o valor da capacidade maxima de adsorcao em [M /M ]\&. "
.ti -1c
.RI "Real \fBK1\fP () const"
.br
.RI "Funcao que retorna o coeficiente associado a constante de equilibrio de \fBElovich\fP\&. "
.ti -1c
.RI "void \fBQmax\fP (const Real &_qmax)"
.br
.RI "Funcao para alterar a capacidade maxima de adsorcao em [M /M ]\&. \&. "
.ti -1c
.RI "void \fBK1\fP (const Real &_k1)"
.br
.RI "Funcao que altera o coeficiente associado a constante de equilibrio de \fBElovich\fP\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &_c, const Real &_temp) const override"
.br
.RI "Funcao que calcula a quantidade de sorcao no equilibrio\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &_c) const override"
.br
.ti -1c
.RI "virtual std::unique_ptr< \fBIsotherm\fP > \fBCloneImplementation\fP () const override"
.br
.in -1c
.SS "Outros membros herdados"
.SH "Descrição detalhada"
.PP 
Classe com as equacoes da isoterma de \fBElovich\fP 

Isoterma com dois parametros, $ K_1 $ e $ q_{max} $, cujas formulas sao as seguintes: \\begin{align} Q_e (C_e) &= q_{max} \\theta \\hfil \\\\ \\\\ \\theta &= K_1 C_e e^{-\\theta} \\end{align} Um artigo de referencia pode ser encontrado \fCaqui\fP\&. 
.PP
\fBAutores\fP
.RS 4
Iasmim Barboza Storck 
.PP
Lara Botelho Brum 
.PP
Luan Rodrigues Soares de Souza 
.PP
Joao Flavio Vieira de Vasconcellos 
.RE
.PP
\fBVersão\fP
.RS 4
1\&.0 
.RE
.PP
\fBData\fP
.RS 4
2022 
.RE
.PP
\fBBug\fP
.RS 4
Nao ha bugs conhecidos\&.
.RE
.PP
.PP
\fBCopyright\fP
.RS 4
GNU Public License\&. << NOTE
.RE
.PP
Defines a class for stuff\&. 
.PP
\fBParâmetros do <em>template</em>\fP
.RS 4
\fI\fBElovich\fP\fP TTipo importante\&. 
.RE
.PP

.SH "Construtores e Destrutores"
.PP 
.SS "Elovich::Elovich ()\fC [default]\fP"

.PP
Construtora default\&. Uso: 
.PP
.nf
Elovich  var;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros para esta funcao 
.RE
.PP

.SS "Elovich::Elovich (const \fBElovich\fP & _orig)\fC [default]\fP"

.PP
Construtora de copia\&. Uso: 
.PP
.nf
Elovich  var;
Elovich  var1(var);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBElovich\fP original\&. 
.RE
.PP

.SS "Elovich::Elovich (const Real & _qmax, const Real & _k1)"

.PP
Construtora com os parametros que definem a isoterma de \fBElovich\fP\&. Uso: 
.PP
.nf
Real qmax(1\&.0);
Real k1(1\&.0);            
Elovich var1(qmax, k1);              

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Capacidade maxima de adsorcao\&. 
.br
 
.br
\fI_k1\fP Constante da isoterma de \fBElovich\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.br
\fI_k1\fP <= 0\&. 
.RE
.PP

.SH "Funções membros"
.PP 
.SS "virtual std::unique_ptr< \fBIsotherm\fP > Elovich::CloneImplementation () const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Implementa \fBIsotherm\fP\&.
.SS "Real Elovich::K1 () const\fC [inline]\fP"

.PP
Funcao que retorna o coeficiente associado a constante de equilibrio de \fBElovich\fP\&. Uso: 
.PP
.nf
Elovich  var1(q1, k1);              
Real k2 = var1\&.K1();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do coeficiente associado a constante de equilibrio de \fBElovich\fP\&. 
.br
 
.RE
.PP

.SS "void Elovich::K1 (const Real & _k1)\fC [inline]\fP"

.PP
Funcao que altera o coeficiente associado a constante de equilibrio de \fBElovich\fP\&. Uso: 
.PP
.nf
Elovich  var1(q1, k1);              
Real k2(2\&.0);
var1\&.K1(k2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k1\fP Novo valor do coeficiente associado a constante de equilibrio de \fBElovich\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k1\fP <= 0\&. 
.RE
.PP

.SS "\fBElovich\fP & Elovich::operator= (const \fBElovich\fP & _orig)\fC [default]\fP"

.PP
Sobrecarga do operador =\&. Uso: 
.PP
.nf
Elovich  var1(q1, k2);              
Elovich  var2 = var1;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBElovich\fP original\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Copia de _orig\&. 
.br
 
.RE
.PP

.SS "Real Elovich::Qe (const Real & _c) const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Reimplementa \fBIsotherm\fP\&.
.SS "Real Elovich::Qe (const Real & _c, const Real & _temp) const\fC [override]\fP, \fC [virtual]\fP"

.PP
Funcao que calcula a quantidade de sorcao no equilibrio\&. Uso: 
.PP
.nf
Elovich  var1(q1, k1);              
Real ce(1);
Real qe = var1\&.Qe(ce);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_c\fP Concentracao do soluto\&. 
.br
\fI_temp\fP Variavel nao utilizada neste modelo e que pode ser omitida\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da quantidade de sorcao no equilibrio\&. 
.br
 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_c\fP < 0\&. 
.RE
.PP

.PP
Implementa \fBIsotherm\fP\&.
.SS "Real Elovich::Qmax () const\fC [inline]\fP"

.PP
Funcao que informa o valor da capacidade maxima de adsorcao em [M /M ]\&. Uso: 
.PP
.nf
Elovich  var1(q1, k1);              
Real q1 = var1\&.Qmax();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do coeficiente de distribuicao em [M /M ]\&. 
.RE
.PP

.SS "void Elovich::Qmax (const Real & _qmax)\fC [inline]\fP"

.PP
Funcao para alterar a capacidade maxima de adsorcao em [M /M ]\&. \&. Uso: 
.PP
.nf
Elovich  var1(q1, k1);              
Real q2(3\&.0);
var1\&.Qmax(q2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Novo valor da capacidade maxima de adsorcao em [M /M ]\&. \&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.br
 
.RE
.PP


.SH "Autor"
.PP 
Gerado automaticamente por Doxygen para Isotherm++ a partir do código-fonte\&.


IsothermLib/src/Isotherm/TwoParameters/Elovich.cpp
//==============================================================================
// Name        : Elovich.cpp
// Authors     : Lara Botelho Brum
//               Luan Rodrigues Soares de Souza
//               Joao Flavio Vieira de Vasconcellos
//               Iasmim Barboza Storck
// Version     : 1.0
// Description : Classe com as equacoes da isoterma de Elovich
//
// Copyright   : Copyright (C) <2022>  Joao Flavio Vasconcellos
//                                      (jflavio at iprj.uerj.br)
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//
//==============================================================================

//==============================================================================
// includes da lib IsothermLib
//==============================================================================

#include <cmath>                    // exp
#include <iostream>                 // std::cout

//==============================================================================
// includes da lib IsothermLib
//==============================================================================

#include <Error/IsoException.h>
#include <Isotherm/TwoParameters/Elovich.h>
#include <Misc/NewtonRaphson.h>


IST_NAMESPACE_OPEN

//==============================================================================
// Variaveis estaticas
//==============================================================================

VecPairString       detailsElovich
    {
        PairString  ( "Qmax"
                    , "Capacidade maxima de adsorcao")
    ,   PairString  ( "K1"
                    , "Constante da isoterma de Elovich")
    };

template<>
VecPairString IsothermTemplate < Elovich >::infoIsotherm = detailsElovich;


//==============================================================================
// Construtora com dois parametros
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "Elovich :: Elovich (const Real&, const Real&)"
Elovich :: Elovich  (   const Real& _qmax
                    ,   const Real& _k1
                    )
                    :   TwoParameters(  _qmax, _k1)
{

    try {
            if (_qmax <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadQmaxLEZero);

            if (_k1 <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK1LEZero);

    } catch (const IsoException& _isoExcept) {

        std::cout << _isoExcept << "\n";
        exit(EXIT_FAILURE);

    }

    setup = true;

    }

//==============================================================================
// Concentracao Qe
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "Real Elovich ::  Qe (const Real&, const Real&) "
Real
Elovich ::  Qe (   const Real& _ce
               ,   const Real& ) const
{

    try {

        if (!setup) throw
            IsoException    (   IST_LOC
                            ,   className()
                            ,   BadCoefficient);

        if (_ce <= 0.0)  throw
                IsoException    (   IST_LOC
                                ,   className()
                                ,   BadCeLEZero);

    } catch (const IsoException& _isoExcept) {

        std::cout << _isoExcept << "\n";
        exit(EXIT_FAILURE);;
    }

    const_cast<Real&>(auxiCe) = _ce * this->K1();

auto fp    = std::bind(&Elovich::FQe, *this, _1);
auto theta = NewtonRaphson (fp, 0.5);
auto value =  theta * Qmax();

    return  (value >= ZERO ? value : 0.0);
}

#undef  __FUNCT__
#define __FUNCT__ "Real  Elovich ::  FQe (const Real&) const"
Real
Elovich ::  FQe (const Real& _theta) const
{

    return _theta - auxiCe * exp (- _theta);

}

IST_NAMESPACE_CLOSE
